cdap-client-tests/src/test/java/co/cask/cdap/client/PreferencesClientTestRun.java

```diff
@@ -130,6 +134,17 @@ public void testProgramAPI() throws Exception {
     }
   }
 
+  private HttpResponse getServiceResponse(URL serviceURL) throws IOException, InterruptedException {
+    int iterations = 0;
+    HttpResponse response;
+    do {
+      TimeUnit.MILLISECONDS.sleep(50);
+      iterations++;
+      response = HttpRequests.execute(HttpRequest.builder(HttpMethod.GET, serviceURL).build());
```

Comment:
better to return if (response == 200) here and do the sleep after that statement. and the while loop will just be 
``while (iterations < 100);